content/post/deploy-and-manage-pgpool-in-gke/index.md

```diff
@@ -0,0 +1,482 @@
+metadata:
+  name: pg-configuration
+  namespace: demo
+stringData:
+  user.conf: max_connections=400
+```
+Let's save this yaml configuration into `pg-configuration.yaml` 
+Then create the above Secret,
+
+```bash
+kubectl apply -f pg-configuration.yaml
+secret/pg-configuration created
+```
+
+Here, is the yaml of the PostgreSQL CRO we are going to use:
+
+```yaml
+apiVersion: kubedb.com/v1alpha2
+kind: Postgres
+metadata:
+  name: postgres-cluster
+  namespace: demo
+spec:
+  replicas: 3
+  version: "16.1"
+  configSecret:
+    name: pg-configuration
+  storageType: Durable
+  storage:
+    storageClassName: "standard"
+    accessModes:
+      - ReadWriteOnce
+    resources:
+      requests:
+        storage: 1Gi
+  terminationPolicy: WipeOut
+```
+
+Let's save this yaml configuration into `postgres-cluster.yaml` 
+Then create the above PostgreSQL CRO,
+
+```bash
+$ kubectl apply -f postgres-cluster.yaml
+postgres.kubedb.com/postgres-cluster created
+```
+In this yaml,
+* `spec.version` field specifies the version of PostgreSQL. Here, we are using PostgreSQL `version 16.1`. You can list the KubeDB supported versions of PostgreSQL by running `$ kubectl get postgresversions` command.
+* `spec.storage` specifies PVC spec that will be dynamically allocated to store data for this database. This storage spec will be passed to the StatefulSet created by KubeDB operator to run database pods. You can specify any StorageClass available in your cluster with appropriate resource requests.
+* `spec.configSecret` is an optional field that allows users to provide custom configuration for PostgreSQL.
+* And the `spec.terminationPolicy` field is *Wipeout* means that the database will be deleted without restrictions. It can also be "Halt", "Delete" and "DoNotTerminate". Learn More about these in [Termination Policy](https://kubedb.com/docs/latest/guides/postgres/concepts/postgres/#specterminationpolicy).
+
+Let’s check if the server is ready to use,
+
+```bash
+$ kubectl get postgres -n demo postgres
+NAME               VERSION   STATUS   AGE
+postgres-cluster   16.1      Ready    2m7s
+```
+
+### Create Database, User & Grant Privileges
+
+Here, we are going to create a database with a new user and grant all privileges to the database. 
+
+```bash
+$ kubectl exec -it postgres-cluster-0 -n demo -- bash
+Defaulted container "postgres" out of: postgres, pg-coordinator, postgres-init-container (init)
+
+postgres-cluster-0:/$ psql -c "create database test"
+CREATE DATABASE
+
+postgres-cluster-0:/$ psql -c "create role roy with login password '12345'"
+CREATE ROLE
+
+postgres-cluster-0:/$ psql -c "grant all privileges on database test to roy"
+GRANT
+
+postgres-cluster-0:/$ psql test
+psql (16.1)
+Type "help" for help.
+
+test=# GRANT ALL ON SCHEMA public TO roy;
+GRANT
+
+test=# exit
+
+postgres-cluster-0:/$ exit
+exit
+```
+
+#### Create Secret
+
+Now, we'll create a secret that includes the `User` and `Password` that we created as PostgreSQL role above.
```

Comment:
Let's replace something with (PostgreSQL role)

Reply:
Now, we'll create a secret that includes the `User` and `Password` with the newly created role and password above.